### PR TITLE
Add DeploymentTarget/DeploymentTargetClaim/DeploymentTargetClass API example to API doc

### PR DESCRIPTION
**Pull request by jgwest**
#### Description:
- Add DeploymentTarget/DeploymentTargetClaim/DeploymentTargetClass API example to API doc

### CODE REVIEW

**Comment by jgwest** [comment]
/retest
Known CI flake which is being investigated.

File: docs/api.md [hunk 1]
```diff
@@ -600,3 +600,132 @@ spec:
...
+kind: Environment
+metadata:
+  name: prod
+spec:
+  # (...)
+  
+  configuration:
+
+	# Target is used to reference a DeploymentTargetClaim for a target Environment.
+	# The Environment controller uses the referenced DeploymentTargetClaim to access its bounded
```
Comment (chetan-rns) [inline docs/api.md:629]: I see a red tab in the markdown version. Removing the extra tab might fix this issue

Screenshot from 2023-09-05 11-41-35

File: docs/api.md [hunk 2]
```diff
@@ -600,3 +600,132 @@ spec:
...
+    # DeploymentTarget with cluster credential secret.
+    target:
+      deploymentTargetClaim:
+        claimName: staging-dtc # refers to a name of DTClaim in the Namespace
+```
+
+The `DeploymentTargetClaim` API resource is similar to Kubernetes' PersistentVolumeClaim resource. DeploymentTarget is Namespace-scoped:
```
Comment (chetan-rns) [inline docs/api.md:636]:
```suggestion
The `DeploymentTargetClaim` API resource is similar to Kubernetes' PersistentVolumeClaim resource. DeploymentTargetClaim is Namespace-scoped:
```

**Review by chetan-rns — COMMENTED** [review]
Looks good @jgwest. Just a quick indentation and typo correction and then it's good to go.

**Comment by jgwest** [comment]
Thanks @chetan-rns!

**Pull request merged**